lyncs_quda/clover_field.py

```diff
@@ -34,89 +34,109 @@ class CloverField(LatticeField):
      * Only rho is mutable.  To change other params, a new instance should be created
      * QUDA convention for clover field := 1+i ( kappa csw )/4 sigma_mu,nu F_mu,nu (<-sigma_mu,nu: spinor tensor)
      *  so that sigma_mu,nu = i[g_mu, g_nu], F_mu,nu = (Q_mu,nu - Q_nu,mu)/8 (1/2 is missing from sigma_mu,nu)
+     * Apparently, an input to QUDA clover object, coeff = kappa*csw 
+     *  wihout a normalization factor of 1/4 or 1/32 (suggested in interface_quda.cpp) 
     """
 
+    _children = {}
+    
+    def __new__(cls, fmunu, **kwargs):
+        #TODO: get dofs and local dims from kwargs, instead of getting them
+        # from self.shape assuming that it has the form (dofs, local_dims)
+        if isinstance(fmunu, CloverField):
+            return fmunu
+        if not isinstance(fmunu, (numpy.ndarray, cupy.ndarray)):
+            raise TypeError(
+                f"Supporting only numpy or cupy for field, got {type(fmunu)}"
+            )
+        is_clover = False
+        if not isinstance(fmunu, GaugeField):
+            if kwargs.get("is_clover", False):
+                is_clover = True
+                parent = type(fmunu)
+                field = fmunu
+            else:
+                fmunu = GaugeField(fmunu)
+                
+        # Set parent
+        if not is_clover: # not copying from a clover-field array
+            idof = int((fmunu.ncol * fmunu.ndims) ** 2 / 2)
+            prec = fmunu.dtype
+            field = fmunu.backend.empty((idof,) + fmunu.dims, dtype=prec)
+            parent = fmunu.backend.ndarray
+        # Set child
+        if parent in cls._children.keys():
+            child  = cls._children.get(parent)
+        else:
+            child = type(cls.__name__+"ext",(cls, parent), {})
+            cls._children.update({parent: child})
+        obj = field.view(type=child)
```

Comment:
Just wondering if you could call at some point `super().__new__(cls, fmunu, **kwargs)` avoiding some duplicated code...